cypress/integration/regression/main-table-regression.spec.ts

```diff
@@ -0,0 +1,226 @@
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 1')
+    mainTablePage.mainTable.tableRow.getButtonByText('p.Y434=').click()
+    variantDrawerPage.variantDrawer.addNote.click()
+    variantDrawerPage.variantDrawer.fillSpace.type('note addition test')
+    cy.intercept('POST', '/app/ws_tags?ds=PGP3140_wgs_panel_hl&rec=**').as(
+      'addNote',
+    )
+    variantDrawerPage.variantDrawer.saveNote.click()
+    cy.wait('@addNote', { timeout: Timeouts.TwentySecondsTimeout })
+  })
+
+  it('should filter main table by custom tag | step 6', () => {
+    mainTablePage.visit('/ws?ds=PGP3140_wgs_panel_hl')
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addTag.first().click()
+    mainTablePage.mainTable.searchFilter.type('CustomTag')
+    mainTablePage.mainTable.checkbox.variantMainTableCheckbox('CustomTag')
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 0')
+  })
+
+  it('should turn off Gene column | step 7', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.type('Gene')
+    mainTablePage.mainTable.columnSwitch.eq(1).click({ force: true })
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(7)
+  })
+
+  it('should turn on Gene column | step 8', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.type('Gene')
+    mainTablePage.mainTable.columnSwitch.eq(1).click({ force: true })
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(7)
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.clear()
+    mainTablePage.mainTable.searchColumn.type('Gene')
```

Comment:
I'd create enum for the existent columns. Because if one of them will be renamed you need to change a lot.

Reply:
Not sure if I understood you right but it's not supposed to change